File: src/main/java/orderbook/orderbook/commands/AbstractIdOrderBookCommand.java
package orderbook.orderbook.commands;

import orderbook.orderbook.OrderManager;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.List;
import java.util.UUID;

import static orderbook.orderbook.parameters.parsers.ParseSender.parseSendUUID;

public abstract class AbstractIdOrderBookCommand extends AbstractOrderBookCommand {

    public AbstractIdOrderBookCommand(OrderManager om) {
        super(om);
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if(args.length == 0) {
            return Collections.emptyList();
        } else if (args.length == 1) {
            return tabCompleteID(args[0]);
        } else {
            return Collections.emptyList();
        }
    }

    protected UUID parseArgs(String[] args, CommandSender sender) {
        if( args.length < 1 ){
            return null;
        }

        return parseSendUUID(args[0], sender);
    }
}

File: src/main/java/orderbook/orderbook/commands/AbstractIdOrderBookCommand.kt
package orderbook.orderbook.commands

import orderbook.orderbook.OrderBook
import orderbook.orderbook.parameters.parsers.parseSendUUID
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import java.util.*

abstract class AbstractIdOrderBookCommand(plugin: OrderBook) : AbstractOrderBookCommand(plugin) {
    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ): List<String>? = when (args.size) {
        1 -> tabCompleteID(args[0])
        else -> emptyList()
    }

    protected fun parseArgs(args: Array<String>, sender: CommandSender): UUID? = args.firstOrNull()?.let { parseSendUUID(it, sender) }
}